### ISSUE TITLE
ALLOW TEMPORARILY Button Causes Notifier Alert To Pop Up Again

### ISSUE THREAD

**Issue by BADJAG** [wokhansoft/WFN#47]
As stated in the title, clicking the ALLOW TEMPORARILY option for a program causes an IDENTICAL pop-up to be generated again. Clicking the ALLOW TEMPORARILY button a second time causes the alerts to stop. In the System Tray, two shield icons then appear for the rule.

However, if the SKIP option is chosen the second time the alert appears, only one temporary rule (shield icon) is created in the System Tray. The program that was allowed is able to access the internet properly.

In other words, each temporary allow alert shows up twice, though only ONE is required for the requesting program to work correctly.

**Comment by BADJAG** [comment]
Until this issue is fixed, here's a temporary workaround: go to the Task Scheduler (taskschd.msc), open the Notifier task's properties, go to the SETTINGS tab, and change the last setting from "RUN A NEW INSTANCE IN PARALLEL" to "QUEUE A NEW INSTANCE". This prevents multiple simultaneous alerts from crashing the system.

**Comment by wokhan** [comment]
Considering this as fixed (or soon to be if not) since it will either be reproduced while strengthening our tests, either already fixed with all improvements made since.

**Comment by harrwiss** [comment]
Just for info:
Implemented a new more simple solution to handle temp block/allow rules with only 1 shield icon shown. All of the temp rules are removed, when clicking on the shield icon or exiting (not killing 😉 ) Notifier. 

Think the handling could still be improved and I did not look into the particular case raised by the poster. The reason I resorted to implement a simpler solution was, that *block temp* did not work at all (error) and *allow temp* created many shield icons wihtout a tooltip to know for what they are.
